app/screens/channel/header/header.tsx

```diff
@@ -0,0 +1,157 @@
+
+        return (<OtherMentionsBadge channelId={channelId}/>);
+    }, [isTablet, channelId, teamId]);
+
+    const rightButtons: HeaderRightButton[] = useMemo(() => ([{
+        iconName: 'magnify',
+        onPress: () => {
+            DeviceEventEmitter.emit(Navigation.NAVIGATE_TO_TAB, {screen: 'Search', params: {searchTerm: `in: ${searchTerm}`}});
+            if (!isTablet) {
+                popTopScreen(componentId);
+            }
+        },
+    }, {
+        iconName: Platform.select({android: 'dots-vertical', default: 'dots-horizontal'}),
+        onPress: () => true,
+        buttonType: 'opacity',
+    }]), [channelId, isTablet, searchTerm]);
+
+    const onBackPress = useCallback(() => {
+        Keyboard.dismiss();
+        popTopScreen(componentId);
+    }, []);
+
+    const onTitlePress = useCallback(() => {
+        // eslint-disable-next-line no-console
+        console.log('Title Press go to Channel Info', displayName);
+    }, [channelId]);
+
+    let title = displayName;
+    if (isOwnDirectMessage) {
+        title = formatMessage({id: 'channel_header.directchannel.you', defaultMessage: '{displayName} (you)'}, {displayName});
+    }
+
+    let subtitle;
+    let subtitleCompanion;
+    if (memberCount) {
+        subtitle = formatMessage({id: 'channel', defaultMessage: '{count, plural, one {# member} other {# members}}'}, {count: memberCount});
+    } else if (!customStatus || isCustomStatusExpired) {
+        subtitle = formatMessage({id: 'channel.details', defaultMessage: 'View details'});
+    }
+
+    if (memberCount || !customStatus || isCustomStatusExpired) {
+        subtitleCompanion = (
+            <CompassIcon
+                color={changeOpacity(theme.sidebarHeaderTextColor, 0.72)}
+                name='chevron-right'
+                size={14}
+            />
+        );
+    } else if (customStatus && customStatus.text) {
+        subtitleCompanion = (
+            <View style={styles.customStatusContainer}>
+                {Boolean(customStatus.emoji) &&
+                <CustomStatusEmoji
+                    customStatus={customStatus}
+                    emojiSize={13}
+                    style={styles.customStatusEmoji}
+                />
+                }
+                <View style={styles.customStatusText}>
+                    <Text
+                        numberOfLines={1}
+                        ellipsizeMode='tail'
+                        style={styles.subtitle}
+                    >
+                        {customStatus.text}
+                    </Text>
+                </View>
+            </View>
+        );
+    }
+
+    return (
+        <NavigationHeader
+            isLargeTitle={false}
+            leftComponent={leftComponent}
+            onBackPress={onBackPress}
+            onTitlePress={onTitlePress}
+            rightButtons={rightButtons}
+            showBackButton={!isTablet}
+            subtitle={subtitle}
+            subtitleCompanion={subtitleCompanion}
```

Comment:
Not sure if worth it (seems to depend on many things) but subtitleCompanion may use a useMemo.